### DISCUSSION TITLE
Technical Steering Committee (TSC)

### DISCUSSION THREAD

**Discussion by nerdvegas** [AcademySoftwareFoundation/rez#1353] [General]
Hi all,

Let's figure out two things:
* Date for first TSC
* Agenda

I feel like the agenda should involve a fairly high level discussion, my suggestion would be:
* how are we going to run this project (what does governance look like? Who can help?)
* what are our dev priorities - what would we most like to see done first? What adds the most value, but is achievable with constrained resources?

Cheers,
Allan

**Comment by instinct-vfx** [comment]
Sounds like a good start to me. I think we should put the focus especially on how to run the project and how to get the TSC set up so we can get moving. Do we already have any input what is formally needed by the ASWF (if there are any formal requirements that is)?

**Comment by nerdvegas** [comment]
I'm chasing up that question on formal requirements as we speak. I doubt
it'll be anything too onerous, so I say let's go ahead with planning it,
and we'll figure it out once I have that info.

**Comment by JeanChristopheMorinPerso** [comment]
There is https://tac.aswf.io/process/tsc_faq.html for what it's worth.

**Comment by instinct-vfx** [comment]
So here are a bunch of points we need to clarify:

For the TSC:
- How many members should the TSC have?
- How do we get to an initial TSC?
- Meeting cadence
- Chairperson and Secretary

For the more practical side of things i would like to see a discussion around development process / approval process and release process (to enable getting things merged and released without compromising on quality)
Overall direction / focus of development is also an important point.

**Comment by bpabel** [comment]
My two cents

The Python Steering Council has a pretty good model -- https://peps.python.org/pep-0013/

**> How many members should the TSC have?**

There should be an odd number to avoid voting deadlocks -- 3 or 5 is probably sufficient. PSC has 5.

**> How do we get to an initial TSC?**

Preferably with the same method used to maintain a TSC in the future. Elections are a common method. The first decision is to decide who gets to vote. Python has a "core developers" group with voting privileges.

**> Meeting cadence**

Can be left up to the steering committee. Probably at least monthly.

**> Chairperson and Secretary**

Steering committee can make this decision amongst themselves.

**Comment by JeanChristopheMorinPerso** [comment]
All ASWF projects charters can be found at https://github.com/AcademySoftwareFoundation/foundation/tree/main/project_charters. Additionally, each repo should have a `GOVERNANCE.md` file describing more in details how the projects are governed. I dont think any of them have a notion of election and I also don't think elections would work well for the kind of size the rez project is, at least not the same kind of election as the Python Steering Council. I have a feeling the TSC format will be influenced by the ASWF and LF.

**Comment by bpabel** [comment]
@JeanChristopheMorinPerso Thanks. I hadn't seen that. It looks like most projects have a TSC comprised initially of Committers. In some projects, TSC members can be added by open nomination and vote of the TSC. In other projects, a TSC member has to resign and then nominate their own successor.

How many people are currently Committers on rez? Is there a desire to align TSC membership with Committers/Contributors?

**Reply by JeanChristopheMorinPerso** [reply]
>How many people are currently Committers on rez?

Right now @nerdvegas is the repo admin and @instinct-vfx and myself are repo maintainers (as you can see with the little tags we have beside our names when we comment in the GitHub repo). Before the move, I think we had maybe 5-6 (or maybe less) contributors (users with merge access, so committers) but I don't have the list.

>Is there a desire to align TSC membership with Committers/Contributors

The idea is to not have a mismatch, so all TSC voting members should be committers (that is have the contributor role in GitHub). Now, generally speaking, a contributor isn't the same thing as a committer and a committer isn't necessarily a TSC voting member (I think it's up to the projects to decide. But it's kind of weird to have committers not be voting members...).

Also saying TSC membership is confusing. Do you mean voting member or just a simple contributor that attends the TSC meetings or a committer?

**Reply by instinct-vfx** [reply]
It's an important distinction. Everyone can (and by all means should if they are interested!) join the TSC meetings, but that does not make everyone a voting member. I think it's a good idea to strive to have consensus based decision and only formally vote if concesus can not be reached. That said i also think it would be beneficial to the project to have a diverse set of voting members as there are very different user types that deserve to be represented i think. How many i don't know to be honest. As for deadlock it could also be handled by the Chairperson being the deciding factor in said cases? Not to overcomplicate things though.

**Reply by maxnbk** [reply]
> > How many people are currently Committers on rez?
> 
> Right now @nerdvegas is the repo admin and @instinct-vfx and myself are repo maintainers (as you can see with the little tags we have beside our names when we comment in the GitHub repo). Before the move, I think we had maybe 5-6 (or maybe less) contributors (users with merge access, so committers) but I don't have the list.
> 
I was one of the other ones. We were labeled as "Contributor", which gave us merge/release-privilege at the time, though I don't know that anyone other than @nerdvegas actually pulled the trigger on a release, purely out of an abundance of caution, and I think that there may have been a mechanism related to pull-request approval, but I'm not clear on that detail. I was getting ready to pull the trigger on a release or two myself just to save Allan some time, but then the ASWF stuff cropped up, so it became less important. (And I switched jobs, which is part of why I haven't been as active in this chat thus far, but with full intent to return and contribute some things!)

If I had to guess who the other 2 or 3 were, it was probably some mixture of @lambdaclan  , @bfloch , and/or @pmolodo , each being amongst the top-numerical-commit-contributors to rez for some arbitrarily decent stretch of time.

**Reply by instinct-vfx** [reply]
Good to see you post here @maxnbk ! @lambdaclan was mostly contributing as part of being on our team here. I am unsure how much interest he has nowadays?

**Reply by bfloch** [reply]
While I haven't contributed recently, I'd be willing to participate if needed, or just have a seat in the back otherwise. I am still devoted to the goals of the project.

**Comment by ColinKennedy** [comment]
Topics like "who should be allowed to vote", meeting cadence, other details I would be happy to see refined over time instead of being set before this meeting initially. Even if the meeting was open to everyone, I don't expect the size to be very big. Could we not start with the very minimalist of setups and add these sorts of evolutions after the meetings grow to a certain size?

**Reply by instinct-vfx** [reply]
The goal is not to set these prior to the meeting, only to gather points to discuss in the first meeting. Currently from what Allan posted i would expect the meeting to be a mix of topics around forming the TSC and also already actual discussion that would happen in the TSC meetings once established. I totally agree that neither should block the other and we should try and get going.

**Reply by instinct-vfx** [reply]
P.S. i am unsure how many of the people that voiced their interest are actually interested in being a "formal" member of the TSC as opposed to being in the meeting - which everyone is invited to!

**Reply by ColinKennedy** [reply]
> The goal is not to set these prior to the meeting, only to gather points to discuss in the first meeting. 

Alright, it's good to hear that. I wondered if that was the case but wanted to double check. Cheers!

**Comment by maxnbk** [comment]
Most of the issues that I think ought to be up for discussion in a first-TSC have been mentioned already.

The main thing on my mind that I think is not-yet-mentioned is how much help rez can get from the ASWF in terms of developer resources, and how the priorities for which things are deemed most-important get determined.

As for scheduling: It would seem to me that the most-important focus for a first-meeting-timeslot would be getting as many people as possible, prioritized towards active/interested maintainers/contributors, into the same meeting room, and to have a designated notes-taker for that first meeting, minimally.

I would be happy to volunteer to take minutes/notes for that first meeting, pre-supposing that it occurs in a time-slot which I can attend. That is not to be confused with my volunteering to take notes for all TSC meetings - I can easily swing it for some of the first meetings or so, just until certain things and roles are established, candidacies raised, votes cast, etc - However I would not want to assign myself such a role as an ongoing commitment arbitrarily into the future, and I'd ultimately prefer to contribute in other manners anyway. Just an offer to help get started on the right foot and record things for posterity. It's also really hard to say things when you're writing down what everyone else is saying. :)

I'd probably recommend that we, from the start, keep meeting minutes/notes PRed into the repository itself, much the way other ASWF projects do.

I think OpenColorIO's template is good:
https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/main/ASWF/meetings/tsc/template.md
which looks like this, when filled out for a given date:
https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/main/ASWF/meetings/tsc/2021-02-08.md

**Reply by instinct-vfx** [reply]
Good points and suggestions. One remark: There is no actual developer resources being provided to the projects as far as i know. See also https://github.com/AcademySoftwareFoundation/foundation/blob/main/overview_deck/ASWF%20High%20Level%20Overview.pdf

My understanding is, that ASWF provides a neutral forum, infrastructure like groups.io, Slack, Github and many others as well as a legal framework, help with CLAs and the likes but no actual developer resources.

**Reply by maxnbk** [reply]
Fair points. Doh. I thought that was the point of the members having to spend "one full-time engineer" as part of their membership fee.

**Comment by fnaum** [comment]
Same here, I'm still quite interested, and want to make it part of my job,
(not like now checking this at 1am :) )

Regarding the note-taking, seems like OCIO stopped using the repo to keep
the notes of the tsc
I might be dreaming, but I think I saw at some point some automatic
transcript of an ASWF meeting. Of course, it won't replace nice brief
bullet points meeting notes but could help to keep meetings documented and
searchable

I saw other projects use https://doodle.com/ to schedule the meetings, it
seems like a paid tool but maybe the AWSF already has that, or have
something else?

Fede

**Reply by instinct-vfx** [reply]
For scheduling the ASWF provides the mailing lists with calendar (the overview for all TSC and TAC meetings is here: https://www.aswf.io/meeting-calendar/)

**Reply by instinct-vfx** [reply]
The actual lists are at lists.aswf.io, Rez already exists here: https://lists.aswf.io/g/rez-discussion

**Reply by instinct-vfx** [reply]
Taking a look @nerdvegas the list has the old logo still :)

**Reply by JeanChristopheMorinPerso** [reply]
Yeah, TSC meetings are supposed to be scheduled using the ASWF provided calendar thing. Also some TSCs just write meeting notes in Google docs without publishing them to GitHub, some also publish to GitHub.

**Reply by maxnbk** [reply]
> Regarding the note-taking, seems like OCIO stopped using the repo to keep the notes of the tsc I might be dreaming, but I think I saw at some point some automatic transcript of an ASWF meeting. Of course, it won't replace nice brief bullet points meeting notes but could help to keep meetings documented and searchable

Yeah, I figure to start things off, just having any kind of notes is good (and important), things can be adapted to whatever the ASWF uses once those are identified and able to be leveraged.

**Comment by nerdvegas** [comment]
Yes. Honestly without any actual resource allocation from ASWF, it all
feels a little tokenistic. Don't get me wrong, it's overall positive, but
damned if there wouldn't be significant further dev on rez done already if
adoption had come with a fulltime engineer. Multiply that across all
adopted projects and you'd be seeing very tangible overall benefit. Combine
that with some sort of licence giving members restricted access to newest
versions perhaps, to provide some sort of motivation to join... maybe.

</rant(ish)>
A

**Comment by nerdvegas** [comment]
update- I'm chatting with Academy now, they will help with the first meeting, which will apparently be largely procedural.

**Comment by JeanChristopheMorinPerso** [comment]
To anyone who subscribed to notifications on this discussion: We'll have some updates pretty soon. Stay tuned!

**Comment by JeanChristopheMorinPerso** [comment]
The TSC has been formed and we had our first meeting today. Mostly procedural.

Members of the TSC are myself, @nerdvegas , @bpabel , @instinct-vfx and @maxnbk .

See https://academysoftwarefdn.slack.com/archives/C0321B828FM/p1660870801111109.
